azkaban-web-server/src/test/java/azkaban/flowtrigger/TriggerInstanceProcessorTest.java

```diff
@@ -47,28 +58,41 @@
   private static ExecutorManager executorManager;
   private static Emailer emailer;
   private static TriggerInstanceProcessor processor;
+  private static EmailMessageCreator messageCreator;
+  private static EmailMessage message;
 
   @BeforeClass
   public static void setup() throws Exception {
+    message = EmailerTest.mockEmailMessage();
+    messageCreator = EmailerTest.mockMessageCreator(message);
     triggerInstLoader = mock(FlowTriggerInstanceLoader.class);
     executorManager = mock(ExecutorManager.class);
     when(executorManager.submitExecutableFlow(any(), anyString())).thenReturn("return");
-    emailer = mock(Emailer.class);
-    doNothing().when(emailer).sendEmail(any(), any(), any());
+    final CommonMetrics commonMetrics = new CommonMetrics(new MetricsManager(new MetricRegistry()));
+    emailer = new Emailer(new Props(), commonMetrics, messageCreator);
     processor = new TriggerInstanceProcessor(executorManager, triggerInstLoader, emailer);
   }
 
+
   private static TriggerInstance createTriggerInstance() {
     final FlowTrigger flowTrigger = new FlowTrigger(
         new CronSchedule("* * * * ? *"),
         new ArrayList<>(),
         Duration.ofMinutes(1)
     );
     final Project proj = new Project(1, "proj");
-    final Flow flow = new Flow("flowId");
+    final Flow flow = new Flow("123");
     flow.addFailureEmails(Lists.newArrayList(EMAIL));
     proj.setFlows(Maps.newHashMap("flowId", flow));
     final List<DependencyInstance> depInstList = new ArrayList<>();
+    depInstList.add(
```

Comment:
Could use Arrays.asList(varargs) instead of new + add? Just slightly cleaner..

Reply:
updated, thanks